cartograph/server/NewMapService.py

```diff
@@ -61,55 +124,31 @@ def on_get(self, req, resp):
+        assert self.map_services['_multi_map']
+
         post_data = falcon.uri.parse_query_string(req.stream.read())
         resp.body = ''
 
         map_name = post_data['name']
-        for c in map_name:
-            assert c in ACCEPTABLE_MAP_NAME_CHARS
-
-        if not os.path.exists(USER_CONF_DIR):
-            os.makedirs(USER_CONF_DIR)
-
-        # Generate dictionary of article names to IDs
-        # TODO: is there a way to do this once (instead of once per POST)?
-        names_path = os.path.join(SOURCE_DIR, 'names.tsv')
-        name_dict = {}
-        with codecs.open(names_path, 'r') as names:
-            names_reader = csv.reader(names, delimiter='\t')
-            for row in names_reader:
-                name = unicode(row[1], encoding='utf-8')
-                name_dict[name] = row[0]
-
-        # Generate list of IDs for article names in user request
-        ids = []
-        for term in post_data['articles'].split('\r\n'):
-            try:
-                ids += [name_dict[term]]  # Attempts to find entry in dict of Articles to IDs
-            except KeyError:
-                # TODO: The following line is for debugging; proper behavior yet to be defined
-                resp.body += 'NO MATCH FOR TERM: %s\n' % (term,)
-
-        # Create the destination directory (if it doesn't exist already)
-        target_path = os.path.join(BASE_PATH, 'user/', map_name)
-        if not os.path.exists(target_path):
-            os.makedirs(target_path)
-
-        # For each of the data files, filter it and output it to the target directory
-        for filename in ['ids.tsv', 'links.tsv', 'names.tsv', 'popularity.tsv', 'vectors.tsv']:
-            filter_tsv(SOURCE_DIR, target_path, ids, filename)
-
-        # Generate a new conf file
-        with open('./data/conf_template.txt', 'r') as conf_template_file:
-            conf_template = string.Template(conf_template_file.read())
-        config_filename = '%s.txt' % pipes.quote(map_name)
-        config_path = os.path.join(USER_CONF_DIR, config_filename)
-        with open(config_path, 'w') as config_file:
-            config_file.write(conf_template.substitute(name=map_name))
-
-        # Build from the new conf file
-        os.system("CARTOGRAPH_CONF=""%s"" PYTHONPATH=$PYTHONPATH:.:./cartograph luigi --module cartograph ParentTask" % config_path)
+        articles = post_data['articles'].split('\r\n')
+
+        # Prevent adding a map with the same name as a currently-served map
+        # This will prevent adding user-generated maps with the same names as
+        # active non-user-generated maps, e.g. "simple" or "en"
+        assert map_name not in self.map_services.keys()
+
+        bad_articles = gen_data(map_name, articles)
+        config_path = gen_config(map_name)
+
+        # Build from the new config file
+        os.system("CARTOGRAPH_CONF=""%s"" PYTHONPATH=$PYTHONPATH:.:./cartograph luigi --module cartograph ParentTask --local-scheduler" % config_path)
```

Comment:
Check exit code... I think you'll need to use subprocess instead. Also (if possible) it would be ideal for the build log to go to a file in the map directory.